scripts/create_burndown.py

```diff
@@ -17,211 +17,402 @@
-	       xytext=(x_days[0], 0.5 - tasks_done_extra), textcoords='data',
-	       arrowprops=dict(arrowstyle="<|-|>", connectionstyle="arc3", color='green')
-	      )
-
-  y_text = - tasks_done_extra / 2
-  plt.text(0.4, y_text, str(int(tasks_done_extra)) + " extra",
-	   rotation='vertical', verticalalignment='center', color='green'
-	  )
-  plt.text(0.7, y_text, "tasks done",
-	   rotation='vertical', verticalalignment='center', color='green'
-	  )
 
+# Parse arguments and/or print help
+def argParser():
+  epilog = "Look at https://github.com/openSUSE/trollolo for details"
+  description = "Generates Scrum Burndown Chart from YAML file"
+  
+  parser = argparse.ArgumentParser(epilog=epilog, description=description)
+  parser.add_argument('sprint', metavar='NUM', help='Sprint Number')
+  parser.add_argument('--output', help='Location of data to process')
+  parser.add_argument('--no-tasks', action='store_true', help='Disable Tasks line in the chart', default=False)
+  parser.add_argument('--with-fast-lane', action='store_true', help='Draw line for Fast Lane cards', default=False)
+  parser.add_argument('--verbose', action='store_true', help='Verbose Output', default=False)
+  parser.add_argument('--no-head', action='store_true', help='Run in headless mode', default=False)
+  args = parser.parse_args()
+  
+  if args.output:
+    os.chdir(args.output)
+  
+  if args.verbose:
+    print args
+
+  return args
+
+
+
+class BurndownData:
+  "Store burndown data parsed from YAML file"
+
+  # Constructor
+  def __init__(self, args):
+    burndown = self.openYAML(args.sprint)
+    self.getSprintData(burndown)
+
+  # Open burndown yaml file
+  def openYAML(self, sprint_number):
+    with open('burndown-data-' + sprint_number + '.yaml', 'r') as f:
+      burndown = yaml.load(f)
+    return burndown
+  
+  # Get data from burndown yaml file
+  def getSprintData(self, burndown):
+    current_day = 1
+  
+    self.sprint_number = burndown["meta"]["sprint"]
+    self.weekend_lines = burndown["meta"]["weekend_lines"]
+    self.total_days = burndown["meta"]["total_days"]
+    self.extra_day = 0
+    self.current_day = 1
+    self.days = []
+    self.extra_days = []
+    self.open_story_points = []
+    self.total_story_points = []
+    self.bonus_story_points_done = []
+    self.open_tasks = []
+    self.total_tasks = []
+    self.bonus_tasks_done = []
+    self.bonus_tasks_day_one = 0
+    self.x_fast_lane = []
+    self.y_fast_lane = []
+    self.total_fast_lane = []
+    self.max_story_points = 0
+  
+    for day in burndown["days"]:
+      self.days.append(current_day)
+      self.open_story_points.append(day["story_points"]["open"])
+      self.total_story_points.append(day["story_points"]["total"])
+      self.open_tasks.append(day["tasks"]["open"])
+      self.total_tasks.append(day["tasks"]["total"])
+      self.max_story_points = max(self.max_story_points, day["story_points"]["total"])
```

Comment:
I would separate processing data and reading it from the YAML. Calculating the maximum story points is different from reading the raw data. So I would move all the code which directly just reads data to the `readYAML` function and have dedicated functions for calculating the data needed for the plot.

In this case it might even make sense to have a separate function `maxStoryPoints`, which calculates the maximum value and returns it, so that the calling code can use it or store it.